cli/lib/nf3.mjs

```diff
@@ -255,35 +261,64 @@ class Nf3 {
   }
 
   /**
-  Method for signing and submitting an Ethereum transaction to the
+  Method for signing an Ethereum transaction to the
   blockchain.
   @method
   @async
   @param {object} unsignedTransaction - An unsigned web3js transaction object.
   @param {string} shieldContractAddress - The address of the Nightfall_3 shield address.
   @param {number} fee - the value of the transaction.
   This can be found using the getContractAddress convenience function.
+  @returns {Oject} Signed transaction.
+  */
+  async _signTransaction(unsignedTransaction, contractAddress, fee) {
+    let tx;
+    let signed;
+
+    await this.nonceMutex.runExclusive(async () => {
+      // estimate the gasPrice
+      const gasPrice = await this.estimateGasPrice();
```

Comment:
The calls to `estimateGasPrice` and `estimateGas` can be moved outside of the mutex, because they don't share state between threads, and as more operations inside a mutex more time it takes for releasing the lock then slow down the whole operation.

Reply:
I’m not 100% sure. I left it out and then we had unordered transactions. So, I prefer to have this like this.